src/main/scala/exercise2/Queue.scala

```diff
@@ -0,0 +1,34 @@
+package exercise2
+
+import scala.util.Try
+
+trait QueueLike[T] {
+  def enqueue(elem: T): QueueLike[T]
+
+  def dequeue(): Try[QueueLike[T]]
+
+  def front(): Option[T]
+
+  def isEmpty: Boolean
+}
+
+class Queue[T](
```

Comment:
A trick one can use is to have the reversing of `in` in the constructor of the queue. This way it's guaranteed that the queue will have at least one element in out (if it has any elements at all) and one will need less checks for that in `dequeue` and `front`.